### PR TITLE
Add Racket Support

### PR DESCRIPTION
**Pull request by smizell**
This PR adds support for [Racket](https://racket-lang.org/), which has some great tools around writing and publishing.

1. [Scribble](https://docs.racket-lang.org/scribble/) can be used to write any kind of docs. Instead of being a markup language, it's a full programming language.
1. [Pollen](https://docs.racket-lang.org/pollen/) is a publishing system for web books that can also be printed. Like Scribble, it gives full Racket support in the markup, making it powerful. It also supports Markdown.
1. [Frog](https://docs.racket-lang.org/frog/index.html) is a static site generator with blog support.

There are many other writing tools provided by Racket, while Racket also gives the ability to create your own language.

### CODE REVIEW

File: run-build-functions.sh [hunk 1]
```diff
@@ -616,6 +616,20 @@ install_dependencies() {
     rm -rf $GOPATH/src/$GO_IMPORT_PATH
     ln -s /opt/buildhome/repo ${GOPATH}/src/$GO_IMPORT_PATH
   fi
+
+  # Install Racket Dependencies
+  if [ -n info.rkt ]
+  then
+    raco setup
```
Comment (Benaiah) [inline run-build-functions.sh]: The description of `raco setup` at https://docs.racket-lang.org/raco/setup.html is as follows:

> The raco setup command builds bytecode, documentation, executables, and metadata indexes for all installed collections.

Shouldn't this be something like the following?

```suggestion
    raco pkg install --deps search-auto
```

I can go ahead and push that change to your branch as well as rebasing it onto master to get it mergeable and adding cache support if you :+1: this comment. (Just don't want to force push your branch without giving you a heads up first).

Comment (smizell) [reply run-build-functions.sh]: Please do! That would be awesome!

**Comment by mraerino** [comment]
once we land #411 this formula might be enough for you?
https://formulae.brew.sh/formula-linux/minimal-racket

**Comment by smizell** [comment]
@mraerino It feels like it might. Would this allow for cached dependencies as well?

**Comment by smizell** [comment]
Is there anything else you need me to do for this? It looks like it's waiting on a reviewer at the point? Thanks!

**Comment by erquhart** [comment]
@mraerino it sounds like you're saying that the homebrew package you linked can be used for Racket support, and that this PR is then unnecessary - can you confirm?

**Comment by vbrown608** [comment]
Thank you for your contribution @smizell. Over the next couple months, we'll be reviewing our use of this build image and how we decide what's preinstalled for builds. Unfortunately, we won't be adding any more dependencies to the build image in the meantime unless there's an urgent reason to do so.

**Comment by kitop** [comment]
Closing this PR as the it's against xenial branch and the Xenial image is going to be deprecated: https://answers.netlify.com/t/please-read-end-of-support-for-xenial-build-image-everything-you-need-to-know/68239

**Pull request closed without merging**